### PR TITLE
Add RANr generator

### PR DESCRIPTION
**Pull request by JoelPasvolsky**

### CODE REVIEW

**Comment by JoelPasvolsky** [comment]
So why is it I didn't just generalize to a RANk generator? 
image

**Comment by arcondello** [comment]
Can you add the citation for the definition?

File: dimod/generators/random.py [hunk 1]
```diff
@@ -133,3 +133,72 @@ def randint(graph, vartype, low=0, high=1, cls=BinaryQuadraticModel,
...
+    Args:
+        r (int):
+            Order of the RANr problem.
+
+        graph (int/tuple[nodes, edges]/:obj:`~networkx.Graph`):
+            The graph to build the binary quadratic model (BQM) for. Either an
+            integer n, interpreted as a complete graph of size n, or
+            a nodes/edges pair, or a NetworkX graph.
+
+        vartype (:class:`.Vartype`/str/set):
```
Comment (arcondello) [inline dimod/generators/random.py]: RANr problems really only make sense for SPIN-valued BQMs. I think this should just always return SPIN.

Comment (JoelPasvolsky) [reply dimod/generators/random.py]: Done

**Pull request merged**